helpers/helpers.go

```diff
@@ -0,0 +1,133 @@
+package helpers
+
+import (
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"crypto/subtle"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/spf13/viper"
+)
+
+// Config holds the configuration of the service
+var Config Conf
+
+// Conf describes the configuration of the service
+type Conf struct {
+	PathToKey      string
+	Iss            string
+	ExpirationDays int
+	Username       string
+	Password       string
+	S3URL          string
+}
+
+// NewConf read the configuration from the config.yaml file
+func NewConf(conf *Conf) (err error) {
+	viper.SetConfigName("config")
+	viper.AddConfigPath(".")
+	viper.AutomaticEnv()
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	viper.SetConfigType("yaml")
+	if viper.IsSet("configPath") {
+		configPath := viper.GetString("configPath")
+		splitPath := strings.Split(strings.TrimLeft(configPath, "/"), "/")
+		viper.AddConfigPath(path.Join(splitPath...))
+	}
+
+	if viper.IsSet("configFile") {
+		viper.SetConfigFile(viper.GetString("configFile"))
+	}
+
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			log.Println("No config file found, using ENVs only")
+		} else {
+			return err
+		}
+	}
+
+	if !viper.IsSet("global.iss") || !viper.IsSet("global.pathToKey") ||
+		!viper.IsSet("global.uppmaxUsername") || !viper.IsSet("global.uppmaxPassword") ||
+		!viper.IsSet("global.s3url") {
+		return fmt.Errorf("Required fields not found in configuration")
+	}
+
+	conf.Iss = viper.GetString("global.iss")
+	conf.PathToKey = viper.GetString("global.pathToKey")
+	conf.Username = viper.GetString("global.uppmaxUsername")
+	conf.Password = viper.GetString("global.uppmaxPassword")
+	conf.S3URL = viper.GetString("global.s3url")
```

Comment:
All these are mandatory, ensure they are not empty and that the key is readable.